src/storage/redis_metadata.h

```diff
@@ -336,3 +338,27 @@ class HyperLogLogMetadata : public Metadata {
 
   EncodeType encode_type = EncodeType::DENSE;
 };
+
+class TDigestMetadata : public Metadata {
+ public:
+  uint64_t compression;
+  uint64_t capacity;
+  uint64_t unmerged_nodes = 0;
+  uint64_t merged_nodes = 0;
+  uint64_t total_weight = 0;
+  uint64_t merged_weight = 0;
+  double minimum = std::numeric_limits<double>::infinity();
+  double maximum = -1 * std::numeric_limits<double>::infinity();
+  uint64_t total_observations = 0;
+  uint64_t merge_times = 0;
+
+  explicit TDigestMetadata(uint64_t compression, uint64_t capacity, bool generate_version = true)
+      : Metadata(kRedisTDigest, generate_version), compression(compression), capacity(capacity) {}
+  explicit TDigestMetadata(bool generate_version = true) : TDigestMetadata(0, 0, generate_version) {}
+  void Encode(std::string *dst) const override;
+  rocksdb::Status Decode(Slice *input) override;
+
+  uint64_t TotalNodes() const { return merged_nodes + unmerged_nodes; }
+
+  double Delta() const { return 1. / static_cast<double>(compression); }
```

Comment:
So delta means compression here?

Reply:
Hi @mapleFU ,

Yes, this delta should be the compression. 😊

Best Regards,
Edward